CorsixTH/Lua/epidemic.lua

```diff
@@ -131,24 +131,29 @@ end
  them into an infected patient too. ]]
 function Epidemic:infectOtherPatients()
   --[[ Can an infected patient infect another patient - taking into account
-   spread factor as defined in the configuration. Patients must be both in the
-   corridors  or in the same room - don't infect through walls.
-   @param patient (Patient) already infected patient
-   @param other (Patient) target to possibly infect
-   @return true if patient can infect other, false otherwise (boolean) ]]
-   local function canInfectOther(patient,other)
-     local can_infect = false
-     if not other.infected and not other.cured and not other.vaccinated
-       and not patient.cured and not patient.vaccinated and not other.is_emergency and
-       (patient.disease == other.disease or other.disease.contagious and not other.diagnosed) and
-       not other.attempted_to_infect
-       and patient:getRoom() == other:getRoom() then
-       can_infect=true
-     end
-     return can_infect
-   end
-
-  local function infect_other(infected_patient,patient)
+  spread factor as defined in the configuration. Patients must be both in the
+  corridors or in the same room - don't infect through walls.
+  @param patient (Patient) already infected patient
+  @param other (Patient) target to possibly infect
+  @return true if patient can infect other, false otherwise (boolean) ]]
+  local function canInfectOther(patient, other)
+    -- Patient is not infectious.
+    if patient.cured or patient.vaccinated then return false end
+
+    -- 'other' is already infected or is going home.
+    if other.infected or other.cured or other.vaccinated then return false end
+    if other.is_emergency then return false end -- Don't interact with emergencies.
+
+    if patient.disease ~= other.disease and
```

Comment:
Comment: -- If the other patient has a different disease OR the other patient's disease cannot be changed